src/ophys_etl/modules/segmentation/merge/characteristic_timeseries.py

```diff
@@ -0,0 +1,278 @@
+"""
+This module contains the code that roi_merging.py uses to find
+the single timeseries characterizing each ROI
+"""
+from typing import List, Tuple, Union, Set, Dict
+import sys
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+import numpy as np
+import multiprocessing
+import multiprocessing.managers
+from ophys_etl.modules.segmentation.merge.utils import (
+    _winnow_process_list)
+
+from ophys_etl.modules.segmentation.\
+    merge.roi_time_correlation import (
+        get_characteristic_timeseries,
+        get_characteristic_timeseries_parallel)
+
+import logging
+
+logger = logging.getLogger(__name__)
+logging.captureWarnings(True)
+logging.basicConfig(level=logging.INFO)
+
+
+class CharacteristicTimeseries(TypedDict):
+    area: float  # tracked so we don't needlessly update after small mergers
+    timeseries: np.ndarray
+
+
+def _update_timeseries_lookup_per_pix(
+        needed_rois:  Union[List[int], Set[int]],
+        sub_video_lookup: Dict[int, np.ndarray],
+        filter_fraction: float,
+        n_processors: int) -> dict:
+    """
+    Method to calculate the characteristic time series of
+    ROIs, using multiprocessing to parallelize each ROI
+    on the pixel level.
+
+    Parameters
+    ----------
+    needed_rois: Union[List[int], Set[int]]
+        List or set containing the ROI IDs of the ROIs whose
+        characteristic timeseries are being calculated
+
+    sub_video_lookup: Dict[int, np.ndarray]
+        Dict mapping ROI ID to sub-videos which have been
+        flattened in space so that their shapes are
+        (ntime, npixels)
+
+    filter_fraction: float
+        The fraction of timesteps (chosen to be the brightest) to
+        keep when doing the correlation.
+
+    n_processors: int
+        Number of processors to invoke with multiprocessing
+
+    Returns
+    -------
+    output: Dict[int, np.ndarray]
+        Maps ROI ID to the characteristic timeseries for that ROI
+    """
+
+    final_output = {}
+    for roi_id in needed_rois:
+        sub_video = sub_video_lookup[roi_id]
+        final_output[roi_id] = get_characteristic_timeseries_parallel(
+                                      sub_video,
+                                      filter_fraction=filter_fraction,
+                                      n_processors=n_processors)
+
+    return final_output
+
+
+def _get_characteristic_timeseries(
+        roi_id_list: List[int],
+        sub_video_lookup: Dict[int, np.ndarray],
+        filter_fraction: float,
+        output_dict: multiprocessing.managers.DictProxy) -> dict:
```

Comment:
for clarity and for testing, could we change these type hints like:
```
output_dict: multiprocessing.managers.DictProxy
```
to
```
output_dict: Union[dict, multiprocessing.managers.DictProxy]
```

Reply:
Done